testcases/GattTestCase.py

```diff
@@ -43,106 +44,137 @@ def test_btp_GATT_CL_GAD_1(self):
         Verify that a Generic Attribute Profile client discovers Primary
         Services in a GATT server.
         """
-
+        stack = self.iut1.stack
         self.verify_skipped(sys._getframe().f_code.co_name)
 
         connection_procedure(self, central=self.iut1, peripheral=self.iut2)
 
-        btp.gattc_disc_prim_svcs(self.iut1,
-                                 self.iut2.stack.gap.iut_addr_get())
-
-        db = GattDB()
-        btp.gattc_disc_prim_svcs_rsp(self.iut1, db)
+        if stack.gatt_cl:
```

Comment:
Maybe it is a good idea to abstract this logic as helper function, like so:
```suggestion
def gattc_disc_prim_svcs(addr1, addr2):
    if stack.gatt_cl:
            btp.gatt_cl_disc_prim_svcs(addr1, addr2)
            stack.gatt_cl.wait_for_prim_svcs()
            db = stack.gatt_cl.db
        else:
            btp.gattc_disc_prim_svcs(addr1, addr2)
            db = GattDB()
            btp.gattc_disc_prim_svcs_rsp(addr1, db)
    return db
```

and then, here do:
db = gattc_disc_prim_svcs(self.iut1, self.iut2.stack.gap.iut_addr_get())

applies to the rest of `if gatt_cl`, too

Reply:
That's a great idea, will apply it

Reply:
@KKopyscinski Helper functions for all GATT procedures are now implemented and used in GATT test cases.
All tests finished with PASS verdict except for one which for some reason randomly fails at connection procedure. I'm looking into this issue right now but this is unrelated to test case handlers that this PR changes

![image](https://github.com/user-attachments/assets/bcce72b5-c306-45cf-89ee-3f0c64649121)